integration-tests/src/test/java/oracle/kubernetes/operator/ITStickySession.java

```diff
@@ -0,0 +1,302 @@
+  private static final String scriptName = "buildDeployAppInPod.sh";
+
+  private static Map<String, String> httpAttrMap;
+
+  private static String httpHeaderFile;
+
+  private static Operator operator;
+  private static Domain domain;
+
+  /**
+   * This method gets called only once before any of the test methods are executed. It does the
+   * initialization of the integration test properties defined in OperatorIT.properties and setting
+   * the resultRoot, pvRoot and projectRoot attributes. It creates an operator, a Weblogic domain
+   * and deploy a web application with persistent-store-type configured to replicated_if_clustered.
+   *
+   * @throws Exception
+   */
+  @BeforeClass
+  public static void staticPrepare() throws Exception {
+    if (!QUICKTEST) {
+      // initialize test properties and create the directories
+      initialize(APP_PROPS_FILE);
+
+      // Create operator1
+      if (operator == null) {
+        logger.info("Creating Operator & waiting for the script to complete execution");
+        operator = TestUtils.createOperator(OPERATOR1_YAML);
+      }
+
+      // create domain
+      if (domain == null) {
+        logger.info("Creating WLS Domain & waiting for the script to complete execution");
+        System.setProperty("LB_TYPE", "VOYAGER");
+        domain = TestUtils.createDomain(DOMAINONPV_WLST_YAML);
+        domain.verifyDomainCreated();
+      }
+
+      httpHeaderFile = BaseTest.getResultDir() + "/headers";
+
+      httpAttrMap = new HashMap<String, String>();
+      httpAttrMap.put("sessioncreatetime", "(.*)sessioncreatetime>(.*)</sessioncreatetime(.*)");
+      httpAttrMap.put("sessionid", "(.*)sessionid>(.*)</sessionid(.*)");
+      httpAttrMap.put("servername", "(.*)connectedservername>(.*)</connectedservername(.*)");
+      httpAttrMap.put("count", "(.*)countattribute>(.*)</countattribute(.*)");
+
+      // Build WAR in the admin pod and deploy it from the admin pod to a weblogic target
+      domain.buildDeployJavaAppInPod(
+          testAppName, scriptName, BaseTest.getUsername(), BaseTest.getPassword());
+
+      // Wait some time for deployment gets ready
+      Thread.sleep(10 * 1000);
+    }
+  }
+
+  /**
+   * Releases k8s cluster lease, archives result, pv directories
+   *
+   * @throws Exception
+   */
+  @AfterClass
+  public static void staticUnPrepare() throws Exception {
+    logger.info("+++++++++++++++++++++++++++++++++---------------------------------+");
+    logger.info("BEGIN");
+    logger.info("Run once, release cluster lease");
+
+    tearDown(new Object() {}.getClass().getEnclosingClass().getSimpleName());
+
+    logger.info("SUCCESS");
+  }
+
+  /**
+   * Use a web application deployed on Weblogic cluster to track HTTP session. In-memory replication
+   * persistence method is configured to implement session persistence. server-affinity is achieved
+   * by Voyager load balancer based on HTTP session information. This test sends two HTTP requests
```

Comment:
Is there a reason why you can test this use case with only Voyager LB ?

Reply:
Yes. Treafik doesn't work due to the bug 28050300 . Pani asked me to use Voyager in P4. Apache should work. If you wanted one with Apache, I can add another java file

Reply:
Voyager is fine, please mention the issue with Traefik and bug in the test comments or description.